### ISSUE TITLE
amdRequire returns module in ES6 format

### ISSUE THREAD

**Issue by ryx** [systemjs/systemjs#1443]
Hi. I am trying to switch a project from requirejs to systemjs but ran into an issue with the format of the returned modules. Our plan is to first replace requirejs with systemjs and then incrementally move the entire codebase to ES6-style imports.

We are using simple AMD-wrappers for global objects to aid in dependency injection. E.g. our `window` wrapper module looks like this:
```
define("gk/globals/window", [], function() {
  return window;
});
```

The above code is expected to directly return the window object when loaded as a dependency of another module (same issue when using require). The following example illustrates how this module is used:
```
define("somemodule", ["gk/globals/window"], function (win) {
  // is expected to output a reference to "window"
  console.log(win);
  // but instead "window" is stored in the "default" property
  console.log(win.default);
});
```

Instead of returning the global `window`, `win` contains an ES6-style module reference with getter/setter and a `default` property, which then points to the global `window` instance.  This appears like a bug to me, because AMD loaders usually don't change the returned module object (at least requirejs doesn't). Can you confirm this is a bug or am I missing some configuration? Thanks in advance.

Best regards,
ryx

**Comment by arty-name** [comment]
It boils down [to this line](https://github.com/systemjs/systemjs/blob/master/dist/system.src.js#L3327):

```javascript
    else if (entry.esmExports && exports !== __global)
      entry.esModule = loader.newModule(getESModule(exports));
``` 

Removing the check for equality to __global solves the issue. I assume we should provide the `gk/globals/window` in a different fashion, so that execution here goes in a different branch.

**Comment by guybedford** [comment]
Thanks, yes this is a bug. I've added https://github.com/systemjs/systemjs/commit/28c84e65c15bb5d35f91256669ad076b2174bf7b for the next release.

**Comment by arty-name** [comment]
As a workaround until this next release we can use following code:

```javascript
System.set("window", System.newModule({default: window, __useDefault: true}))
System.import("window").then(w => console.log(w))
```

**Comment by ryx** [comment]
Awesome, thanks for the quick reply and fix, @guybedford

**Comment by guybedford** [comment]
This was released in 0.19.39.
